### PR TITLE
fix(menu): replace type with interface and add subcomponents to story

### PR DESCRIPTION
**Pull request by David-Cohen35**

### CODE REVIEW

**Review by reywright — APPROVED** [review]

**Comment by reywright** [comment]
Screen Shot 2022-08-02 at 11 32 13 AM

Looks good!

**Review by ivanhdez28 — APPROVED** [review]
lgtm

**Comment by cbedwinotalvaro** [comment]
I don't know if I am missing something but I only see the "asChild" prop, is that right?

image
image
image

**Comment by David-Cohen35** [comment]
> I don't know if I am missing something but I only see the "asChild" prop, is that right?
> 
> image image image

i dont think it is - good catch

**Comment by David-Cohen35** [comment]
im not sure why `ContextMenuTrigger` shows all of the same props as `ContextMenuContent` while the formatting is the same for Menu

File: src/components/ContextMenu/index.tsx [hunk 1]
```diff
@@ -3,12 +3,21 @@ import * as RadixContextMenu from '@radix-ui/react-context-menu';
...
 
-const ContextMenu = ({ children, onOpenChange }: ContextMenuProps): JSX.Element => {
+const ContextMenu = ({
+  children,
+  onOpenChange,
+  ...rest
+}: ContextMenuProps): JSX.Element => {
   return (
-    <RadixContextMenu.Root onOpenChange={onOpenChange} dir="ltr" modal>
+    <RadixContextMenu.Root {...rest} onOpenChange={onOpenChange} dir="ltr" modal>
```
Comment (cbedwinotalvaro) [inline src/components/ContextMenu/index.tsx:20]: is there a reason to do `{...rest}` here? rest is an empty object

Comment (David-Cohen35) [reply src/components/ContextMenu/index.tsx:20]: was just matching the format of `Menu` when we didnt know the issue was the import

**Review by cbedwinotalvaro — APPROVED** [review]

**Pull request merged**